Lib/test/test_argparse.py

```diff
@@ -1934,6 +1934,30 @@ def test_required_subparsers_default(self):
         subparsers.add_parser('run')
         self._test_required_subparsers(parser)
 
+    def test_required_subparsers_no_destination_error(self):
+        parser = ErrorRaisingArgumentParser()
+        subparsers = parser.add_subparsers()
+        subparsers.add_parser('foo')
+        subparsers.add_parser('bar')
+        with self.assertRaises(ArgumentParserError) as excinfo:
+            parser.parse_args(())
+        self.assertRegex(
+            excinfo.exception.stderr,
+            'error: the following arguments are required: {foo,bar}\n$'
```

Comment:
Add a space after comma. Why braces are used?

Reply:
I think this is a notation for alternatives: it means one of foo or bar is required.

Reply:
This is consistent with the rendering of `choices`:

```python
import argparse
parser = argparse.ArgumentParser()
parser.add_argument('--foo', choices=('baz', 'womp'))
parser.parse_args()
```

```
$ python test.py --help
usage: test.py [-h] [--foo {baz,womp}]

optional arguments:
  -h, --help        show this help message and exit
  --foo {baz,womp}
```